src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -0,0 +1,235 @@
+   *
+   * @return true if the service account is set to auto-detect but it can't be fetched from the environment.
+   */
+  public boolean autoServiceAccountUnavailable() {
+    if (connection == null || connection.getServiceAccountFilePath() == null &&
+      connection.isServiceAccountFilePath()) {
+      try {
+        ServiceAccountCredentials.getApplicationDefault();
+      } catch (IOException e) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @param failureCollector  failureCollector to collect the exceptions and failures.
+   * @return GoogleCredentials based on service account
+   */
+  public GoogleCredentials getCredentials(FailureCollector failureCollector) {
+    GoogleCredentials credentials = null;
+    // validate service account
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      credentials =
+        (GoogleCredentials) connection.getCredentials(failureCollector);
+    } else {
+      try {
+        credentials = ServiceAccountCredentials.getApplicationDefault();
+      } catch (IOException e) {
+        failureCollector.addFailure(e.getMessage(), null);
+      }
+    }
+    return credentials;
+  }
+
+  /**
+   * @return Project ID
+   */
+  public String getProject() {
+    if (connection == null) {
+      throw new IllegalArgumentException(
+        "Could not get project information, connection should not be null!");
+    }
+    return connection.getProject();
+  }
+
+  public GCPConnectorConfig getConnection() {
+    return connection;
+  }
+
+  @Nullable
+  public String tryGetProject() {
+    return connection == null ? null : connection.tryGetProject();
+  }
+
+  @Nullable
+  public String getServiceAccount() {
+    return connection == null ? null : connection.getServiceAccount();
+  }
+
+  @Nullable
+  public String getServiceAccountType() {
+    return connection == null ? null : connection.getServiceAccountType();
+  }
+
+  @Nullable
+  public Boolean isServiceAccountFilePath() {
+    return connection == null ? null : connection.isServiceAccountFilePath();
+  }
+
+  /**
+   * Check if service account is a valid one.
+   *
+   * @param failureCollector
+   */
+  public void validateServiceAccount(FailureCollector failureCollector) {
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      GoogleCredentials credentials = getCredentials(failureCollector);
+      if (credentials == null) {
+        failureCollector.addFailure(String.format("Unable to load credentials from %s.",
+              connection.isServiceAccountFilePath() ? connection.getServiceAccountFilePath() : "provided JSON key"),
+            "Ensure the service account file is available on the local filesystem.")
+          .withConfigProperty("serviceFilePath")
+          .withConfigProperty("serviceAccountJSON");
+        throw failureCollector.getOrThrowException();
```

Comment:
Should not throw the exceptions at this time. Usually the pattern followed is - calling all the config validations from `configurePipeline` and `prepareRun` and then throw all at once.

Reply:
All the subsequent validations are based on this credentials. I changed it to validateAndGetServiceAccountCredentials to refrain from calling getCredentials method multiple times

Reply:
You can create a base validate method which calls all the subsequent validation methods. In which you can call this method first and if failureCollector is not empty can skip all the further validations.

Reply:
done